arangod/StorageEngine/TransactionState.cpp

```diff
@@ -320,29 +321,26 @@ Result TransactionState::checkCollectionPermission(std::string const& cname,
   ExecContext const& exec = ExecContext::current();
 
   Result res;
-  // no need to check for superuser, cluster_sync tests break otherwise
-  if (!exec.isSuperuser()) {
```

Comment:
If the user is a superuser then the user should have access to the collection. Therefore removing this short-cut as Superuser is not well-defined.

Reply:
@neunhoef do you understand the comment "no need to check for superuser, cluster_sync tests break otherwise"? That is the magic that breaks them?